core/schema/openai.go

```diff
@@ -108,10 +106,10 @@ type ChatCompletionResponseFormat struct {
 }
 
 type OpenAIRequest struct {
```

Comment:
Should we maybe have the OpenAI and LocalAI requests under `http`? I hardly believe that we are going to re-use the same options around, but maybe you are planning to re-use this with the MQTT work, then would maybe makes sense to actually move it outside from `core/`, but have rather a `schema/` at the top-level package

Reply:
Yeah - I was planning on re-using this for the MQTT work. While I'm envisioning a few different options for MQTT, one of the initial (and most important to me) is something that works on the same messages and objects that the HTTP world does - so that if a backend is not currently available for some reason, the request could be queued up verbatim, and popped off at a later date.